From f6117695ac78ad078a5c441e329947e1ddd17357 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Fri, 17 Jun 2022 13:07:44 +0300
Subject: [PATCH] xbps-src: check for existing packages before syncing chroot
 repodata

This significantly speeds up the package building process for build
scripts etc. that run './xbps-src pkg ...' on a long list of packages
that may already have up-to-date local packages built when configured
with XBPS_PRESERVE_PKGS set.

Link: https://github.com/void-linux/void-packages/pull/37595
---
 xbps-src | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbps-src b/xbps-src
index dfa059927c..4e83490917 100755
--- a/xbps-src
+++ b/xbps-src
@@ -806,6 +806,7 @@ case "$XBPS_TARGET" in
         if [ -n "$CHROOT_READY" -a -z "$IN_CHROOT" ]; then
             chroot_handler $XBPS_TARGET $XBPS_TARGET_PKG
         else
+            check_existing_pkg
             chroot_sync_repodata
             # prevent update_base_chroot from removing the builddir/destdir
             update_base_chroot keep-all-force
-- 
2.36.1

